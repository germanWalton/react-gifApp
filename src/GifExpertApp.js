import React from "react";
import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";


const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Beavis and Butthead"]);

  return (
    <>
      <div className="container">
      <h2 className="p-5 text-center">GifApp</h2>
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((element) => (
          <GifGrid key={element} category={element} />
        ))}
        </ol>
        </div>
    </>
  );
};

export default GifExpertApp;
