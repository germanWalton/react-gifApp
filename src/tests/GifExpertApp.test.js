import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';


describe('pruebas en <GifExpertApp />', () => {
  
  test('debe mostrar el componente correctamente', () => {
  
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  
})

test('debe mostrar la lista de categorias', () => {
  
  const categories = ['Beavis and Butthead', 'Daria']
  
  const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
  
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('GifGrid').length).toBe(categories.length);


})

})