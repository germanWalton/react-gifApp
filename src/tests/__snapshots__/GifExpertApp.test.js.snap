// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="p-5 text-center"
    >
      GifApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <ol>
      <GifGrid
        category="Beavis and Butthead"
        key="Beavis and Butthead"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`pruebas en <GifExpertApp /> debe mostrar la lista de categorias 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h2
      className="p-5 text-center"
    >
      GifApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <ol>
      <GifGrid
        category="Beavis and Butthead"
        key="Beavis and Butthead"
      />
      <GifGrid
        category="Daria"
        key="Daria"
      />
    </ol>
  </div>
</Fragment>
`;
